### DISCUSSION TITLE
Add extra configs in doxyfile generated by doxygen plugin

### DISCUSSION THREAD

**Discussion by jingkaimori** [xmake-io/xmake#4010] [Ideas]
### Is your feature request related to a problem? Please describe.

XMake now simply add source file and project info in generated doxyfile, and there is no way to config extra options such as class graph.

### Describe the solution you'd like

Add a global api `set_doxygen_option` in description scope, which should looks like this:
```lua
set_doxygen_option({
    case_sense_names = true,
    warning = {
        warn_if_undocument = true,
        warn_format = "$file:$line: $text"
        warn_logfile = ""
    },
    generate = {
         html = {
             -- options
         },
         rtf = false
    }
})
```

### Describe alternatives you've considered

Add an option in `xmake doxygen`, which declare a custom doxyfile for this project:
```shell
xmake doxygen --doxyfile ./my/doxyfile
```
xmake should detect if there is duplicate config field in this file and generated file, such as `PROJECT_NAME`, and give proper warning.

**Comment by waruqi** [comment]
xmake does not normally all new api's for specific plugins, it is not very generic.

> Add an option in xmake doxygen, which declare a custom doxyfile for this project:

You can put your custom doxyfile in the root of your project and xmake will use it first and not overwrite it.
